velox/exec/tests/HashJoinTest.cpp

```diff
@@ -206,6 +206,69 @@ TEST_F(HashJoinTest, filter) {
       "((t_k0 % 100) + (u_k0 % 100)) % 40 < 20");
 }
 
+TEST_F(HashJoinTest, joinSidesDifferentSchema) {
+  // In this join, the tables have different schema. LHS table t has schema
+  // {INTEGER, VARCHAR, INTEGER}. RHS table u has schema {INTEGER, REAL,
+  // INTEGER}. The filter predicate uses
+  // a column from the right table  before the left and the corresponding
+  // columns at the same channel number(1) have different types. This has been
+  // a source of crashes in the join logic.
+
+  size_t batchSize = 100;
+
+  auto stringViewGenerator = [](size_t row) {
```

Comment:
This can be simplified like so:

```
std::vector<std::string> strings = {"aaa", "bbb", "ccc", "ddd", "eee"};

... [&] (auto row) { return strings[row % strings.size()];}
```